module/PowerShellEditorServices/Commands/Public/CmdletInterface.ps1

```diff
@@ -78,10 +78,38 @@ function Unregister-EditorCommand {
 }
 
 function Open-EditorFile {
-    param([Parameter(Mandatory=$true)]$FilePaths)
+    [CmdletBinding()]
+    param(
+        [Parameter(Mandatory=$true)]
+        [ValidateNotNullOrEmpty()]
+        $FilePaths,
```

Comment:
This should be singular and in this case I think `Path` is sufficient and consistent with the other PowerShell file manipulation commands.  

Also, I think it would be better to have this command accept pipeline input for the path instead of content.  But that requires some changes to the parameter config - see the Examples\PathProcessingWildcards.ps1 for an example of how to do that. 

With this change, I would be able to do:
```posh
gci *.psd1 -r | Open-EditorFile
```
That seems more useful to me than using the pipeline to overwrite (or create) files with new content.

Reply:
Could do both:

```powershell
function Open-EditorFile {
    [CmdletBinding(PositionalBinding=$false, DefaultParameterSetName='Path')]
    param(
        [Parameter(Mandatory, ParameterSetName='Path', ValueFromPipeline, ValueFromPipelineByPropertyName, Position=0)]
        [Parameter(Mandatory, ParameterSetName='Value', Position=0)]
        [ValidateNotNullOrEmpty()]
        [Alias('FullName')]
        [string] $Path,

        [Parameter(ParameterSetName='Value', ValueFromPipeline)]
        [Parameter(ParameterSetName='Path')]
        [ValidateNotNullOrEmpty()]
        [string] $Value
    )
    begin {
		if ($MyInvocation.ExpectingInput) {
            $valueList = [System.Collections.Generic.List[string]]::new()
        }
    }
    process {
        if ($PSCmdlet.ParameterSetName -eq 'Value') {
            $valueList.Add($Value)
            return
        }

        $psEditor.Workspace.OpenFile($Path)
        if ($Value) {
            $psEditor.GetEditorContext().CurrentFile.InsertText($Value)
        }
    }
    end {
        if ($PSCmdlet.ParameterSetName -eq 'Path') {
            return
        }

        $psEditor.Workspace.OpenFile($Path)
        $psEditor.GetEditorContext().CurrentFile.InsertText($valueList -join [Environment]::NewLine)
    }
}
```

Allows for `gci *.ps1 | Open-EditorFile -Value $x` and `gc .\sourceFile.ps1 | Open-EditorFile destinationFile.ps1`.